apis/management.cattle.io/v3/authn_types.go

```diff
@@ -162,6 +162,28 @@ type GithubConfigApplyInput struct {
 	Enabled      bool         `json:"enabled,omitempty"`
 }
 
+type GoogleOauthConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	AuthConfig        `json:",inline" mapstructure:",squash"`
+
+	OauthCredential          string `json:"oauthCredential,omitempty" norman:"required"`
+	ServiceAccountCredential string `json:"serviceAccountCredential,omitempty"`
+	AdminEmail               string `json:"adminEmail,omitempty"`
+	Hostname                 string `json:"hostname,omitempty" norman:"required"`
+	UserInfoEndpoint         string `json:"userInfoEndpoint" norman:"default=https://openidconnect.googleapis.com/v1/userinfo,required,notnullable"`
```

Comment:
Do we need nocreate on this?

Reply:
Discussed offline and decided it's not needed cc @alena1108 @prachidamle